src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -58,15 +62,17 @@ protected EnvVars buildEnv(AbstractProject<?, ?> job) {
             return env;
         }
 
-        protected static void addRepositories(SCM scm, EnvVars env, Collection<GitHubRepositoryName> r) {
-            if (scm instanceof GitSCM) {
-                GitSCM git = (GitSCM) scm;
-                for (RemoteConfig rc : git.getRepositories()) {
-                    for (URIish uri : rc.getURIs()) {
-                        String url = env.expand(uri.toString());
-                        GitHubRepositoryName repo = GitHubRepositoryName.create(url);
-                        if (repo != null) {
-                            r.add(repo);
+        protected static void addRepositories(Collection<? extends SCM> scms, EnvVars env, Collection<GitHubRepositoryName> r) {
+            for (SCM scm : scms) {
```

Comment:
To minimize the diff, I would recommend moving the `for`-loop into the one caller of this method that will remain after `FromMultiSCM` is deleted.